### PR TITLE
Improve 'recursive-path' documentation.

### LINKED ISSUE #2: recursive-path documentation improvement
- Add example of transform
- Add example of parameterized recursive-path

### PR DESCRIPTION
**Pull request by tasshin**
Should close #2.

### CODE REVIEW

File: List-of-Macros.md [hunk 1]
```diff
@@ -360,12 +360,30 @@ Defines the path that will be associated to the provided name. The name must hav
...
 ```clojure
-=> (let [tree-walker (recursive-path [] p (if-path vector? [ALL p] STAY))]
-     (select tree-walker [1 [2 [3 4] 5] [[6]]]))
+=> (def tree-walker (recursive-path [] p (if-path vector? [ALL p] STAY)))
+#'playground.specter/tree-walker
+;; Get all of the values nested within vectors
+=> (select tree-walker [1 [2 [3 4] 5] [[6]]])
 [1 2 3 4 5 6]
+;; Transform all of the values within vectors
+=> (transform tree-walker inc [1 [2 [3 4] 5] [[6]]])
```
Comment (nathanmarz) [inline List-of-Macros.md:374]: Looks like the result of this expression is missing.

File: List-of-Macros.md [hunk 2]
```diff
@@ -360,12 +360,30 @@ Defines the path that will be associated to the provided name. The name must hav
...
+```
+
+And here is an example of using `recursive-path` with parameters to select and transform:
+
+```clojure
+=> (def map-key-walker (recursive-path [akey] p [ALL (if-path [FIRST #(= % akey)] LAST [LAST p])]))
+#'playground.specter/map-key-walker
+;; Get all the vals for key :aaa, regardless of where they are in the structure
+=> (select (map-key-walker :aaa) {:a {:aaa 3 :b {:c {:aaa 2} :aaa 1}}})
```
Comment (nathanmarz) [inline List-of-Macros.md:384]: Here too.

**Comment by nathanmarz** [comment]
Made a couple comments inline.

**Comment by tasshin** [comment]
Should be fixed now

**Pull request merged**